src/engine/utils/GridUtil.ts

```diff
@@ -7,17 +7,22 @@ export enum GridPlane {
 };
 
 /**
- * Make a plane grid
+ * Make a plane grid centered around origin
+ *
+ * Note that this function takes in HALF-LENGTHS, thus, if you want a grid
```

Comment:
Would it make sense to hide this implemention detail and instead take in full length and width and do the math in the function?

Reply:
We could. It would be a little easier to understand. @JamesWP thoughts? 

In the interface design doc, we had started to talk about half-extents, since the origin point of an object is in its center. But yeah, I think it's more understandable if we can define everything in full lengths (and make it obvious through naming)

Reply:
Oh I see the dilemma now since things like the robot would move from its center point. I can see why we'd want to be consistent. I'm okay to leave it in half lengths.

Reply:
Since this is also an internal utility function, it can stay in whatever lengths we want. The external configuration can be in full-lengths and we'll just do the conversions where appropriate